### PR TITLE
ENH: Dual-acting valve

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Valve with no default position. When losing air/power, it stays where it is. It requires active control to open or close. This has implications for the interlocking.

Icon so far:
image

## Motivation and Context

Three of these valves are in RIX
https://jira.slac.stanford.edu/browse/LCLSPC-338

## How Has This Been Tested?

Interactively only

## Where Has This Been Documented?

Added a page detailing my journey to make a new widget
Add the new widget to the general widget docs

### CODE REVIEW

**Comment by ZLLentz** [comment]
This doesn't need a prompt review or a prompt merge, and it doesn't need to be in the next tag. I may want to do more work on this to clean up the code and we may as a group decide to change the icon still.

**Comment by ZLLentz** [comment]
I'm going to do:
- some minor code cleanup
- docs additions for the new valve type in all the places where all the valves are documented

and then ask for some reviewers once that's taken care of, with the intention of merging

**Comment by ZLLentz** [comment]
One more known thing to do here- looks like this widget and some others aren't fully created in the test suite

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.4.0
```
Comment (klauer) [inline .pre-commit-config.yaml:5]: Would recommend running a `$ pre-commit autoupdate` since you're adding this

File: docs/source/addwidget.rst [hunk 2]
```diff
@@ -0,0 +1,151 @@
+==========================
+Adding a New Vacuum Widget
```
Comment (klauer) [inline docs/source/addwidget.rst:2]: These docs are fantastic

File: pcdswidgets/icons/demo/__main__.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
+"""
+Show an icon. Useful for development.
```
Comment (klauer) [inline pcdswidgets/icons/demo/__main__.py:2]: Very handy utility 👍

File: pcdswidgets/icons/valves.py [hunk 4]
```diff
@@ -356,3 +357,77 @@ def draw_icon(self, painter):
...
+        return self._interlock_brush
+
+    @interlockBrush.setter
+    def interlockBrush(self, new_brush):
+        if new_brush != self._interlock_brush:
+            self._interlock_brush = new_brush
+            self.update()
+
+    def draw_icon(self, painter):
+        path = QPainterPath(QPointF(0, 0.3))
```
Comment (klauer) [inline pcdswidgets/icons/valves.py:386]: https://en.wikipedia.org/wiki/Turtle_graphics

The output looks good, so I'll assume this is all good

**Review by klauer — APPROVED** [review]
I believe you've gone beyond due diligence to "add a dual-acting valve" in this PR, what with the extensive documentation, testing tools, and such.

In summary, this LGTM.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ZLLentz) [reply .pre-commit-config.yaml:5]: I had no idea this was a thing

File: pcdswidgets/icons/valves.py (hunk 4 above)
Comment (ZLLentz) [reply pcdswidgets/icons/valves.py:386]: most of this was copied from the other valve widgets

**Review by tangkong — APPROVED** [review]
Looks great to me.  Makes me wonder what procedure is for adding other widgets, and if it won't be similar.  

This PR is also quite the precedent for other widget-adding PR's

**Comment by ZLLentz** [comment]
Thanks for the look-overs! I'm going to merge and tag because tags are easy when you don't have a releases.rst page

**Pull request merged**